app/components/progressive_image/progressive_image.js

```diff
@@ -37,23 +37,19 @@ export default class ProgressiveImage extends PureComponent {
         this.subscribedToCache = true;
 
         this.state = {
-            intensity: null,
+            intensity: new Animated.Value(80),
             thumb: null,
             uri: null,
         };
     }
 
-    componentWillMount() {
-        const intensity = new Animated.Value(80);
-        this.setState({intensity});
+    componentDidMount() {
         this.load(this.props);
     }
 
-    componentWillReceiveProps(props) {
-        this.load(props);
-    }
-
     componentDidUpdate(prevProps, prevState) {
+        this.load(this.props);
```

Comment:
you should compare the prevProps with this.props here, to avoid loading if not needed

Reply:
This behaviour is basically the same as it was previously, although it's actually better because it only runs after a render as opposed to any time the props might have changed.

I can fix it if you want, but I didn't do it on the first pass since I wasn't sure what should be changing for us to call this.

Reply:
Previously it was running on props change, now it runs even with state changes, and apparently that component updates while the animation happens

Reply:
It was actually ran even when the props don't change as long as the parent is re-rendered. For connected components, I think that might include whenever any store change occurs>

I'll look a bit more into getting this to only run when the important parts of the props change though

Reply:
I updated this and removed some dead code in the `load` method